File: back-end/finanCertoBack/src/main/java/com/finanCerto/finanCertoBack/transacao/Transacao.java
package com.finanCerto.finanCertoBack.transacao;

import com.finanCerto.finanCertoBack.categoria.Categoria;
import com.finanCerto.finanCertoBack.conta.Conta;
import com.finanCerto.finanCertoBack.transacao.dto.TransacaoCadastroDto;
import com.finanCerto.finanCertoBack.usuario.Usuario;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.IdGeneratorType;

import java.time.LocalDateTime;

@Entity
@Table(name = "tb_transacao")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Transacao {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
    private double valor;
    @Column(name = "date", nullable = false)
    private LocalDateTime date;
    private String descricao;
    @Enumerated(EnumType.STRING)
    private Tipos tipo;
    @ManyToOne
    @JoinColumn(name = "conta_id")
    private Conta conta;
    @ManyToOne
    @JoinColumn(name = "categoria_id")
    private Categoria categoria;
    @ManyToOne
    @JoinColumn(name = "usuario_id")
    private Usuario usuario;

    public Transacao(TransacaoCadastroDto dto, Usuario usuario, Categoria categoria, Conta conta) {
        this.descricao = dto.descricao();
        this.valor = dto.valor();
        this.tipo = dto.tipo();
        this.usuario = usuario;
        this.date  = dto.data();
        this.categoria = categoria;
        this.conta = conta;

    }

}

File: front-end/src/types/transacao.ts
export enum TransacaoTipos {
  RECEITA = 'RECEITA',
  DESPESA = 'DESPESA'
}

export interface TransacaoCadastroDto {
  valor: number;
  data: string; // Formato ISO string para LocalDateTime do back-end
  descricao: string;
  tipo: TransacaoTipos;
  nomeConta: string;
  nomeCategoria: string;
  token: string;
}

export interface Categoria {
  id: number;
  nome: string;
  tipo: string;
}

export interface Conta {
  id: number;
  nome: string;
  saldoInicial: number;
  tipos: string;
}

export interface Usuario {
  id: number;
  email: string;
  nome: string;
}

export interface Transacao {
  id?: number;
  valor: number;
  date?: string;
  descricao: string;
  tipo: TransacaoTipos;
  categoria?: Categoria;
  conta?: Conta;
  usuario?: Usuario;
}
